src/Host/Client/Impl/Session/RSession.cs

```diff
@@ -203,14 +203,13 @@ internal sealed class RSession : IRSession, IRCallbacks {
 
         public async Task CancelAllAsync(CancellationToken cancellationToken = default(CancellationToken)) {
             using (_disposeToken.Link(ref cancellationToken)) {
-                var cancelTask = _host.CancelAllAsync(cancellationToken);
+                await _host.CancelAllAsync(cancellationToken);
 
                 var currentRequest = Interlocked.Exchange(ref _currentRequestSource, null);
                 var exception = new OperationCanceledException();
                 currentRequest?.TryCancel(exception);
-                ClearPendingRequests(exception);
 
-                await cancelTask;
+                ClearPendingRequests(exception);
```

Comment:
Do you really need to call ` ClearPendingRequests` after `CancelAllAsync`? I thought it was about `currentRequest?.TryCancel(exception)`